File: java-client/client-app/src/main/java/com/lemondelila/client/game/room/browser/model/PublicRoomSummary.java
package com.lemondelila.client.game.room.browser.model;

public record PublicRoomSummary(int id,
                                String name,
                                String gameType,
                                String status,
                                int maxPlayers,
                                int playersCount,
                                int botsCount,
                                String ownerUsername) {
    @Override
    public String toString() {
        String owner = ownerUsername == null || ownerUsername.isBlank() ? "?" : ownerUsername;
        int current = playersCount + botsCount;
        return "#" + id + " " + name + " (" + gameType + ") " + current + "/" + maxPlayers + " - " + status + " - " + owner;
    }
}

File: client-win/client-win/Modules/Game/Models/PublicRoomSummary.cs
namespace client_win.Modules.Game.Models;

public sealed class PublicRoomSummary
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string GameType { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public int MaxPlayers { get; set; }
    public int PlayersCount { get; set; }
    public int BotsCount { get; set; }
    public string OwnerUsername { get; set; } = string.Empty;

    public override string ToString() => $"{Name} ({PlayersCount}/{MaxPlayers})";
}
